### ISSUE TITLE
Empty attachment id if mail is digitally signed in Outlook desktop

### ISSUE THREAD

**Issue by thowoi** [OfficeDev/office-js#2291]
On Outlook desktop `window.Office.context.mailbox.item.attachments` returns an array of attachments with empty ids and zero size:

```
[
  {
    "attachmentType": "file",
    "contentType": "image/png",
    "id": "",
    "isInline": true,
    "name": "image001.png",
    "size": 0
  }
]
```
Note: Outlook Online works as expected with digitally signed mails.

## Expected Behavior

Attachment id should be present in `window.Office.context.mailbox.item.attachments` array.

## Current Behavior

Attachment id is an empty string ""

## Steps to Reproduce, or Live Example
- Compose a new mail
- Add one or more attachments
- Digitally sign the message [(How to add a digital signature)](https://support.microsoft.com/en-us/office/secure-messages-by-using-a-digital-signature-549ca2f1-a68f-4366-85fa-b3f4b5856fc6)
- Send the message to yourself or some other recipent
- Enumerate the attachments with `window.Office.context.mailbox.item.attachments` and check the id values

## Context
This blocks us from retrieving attachment contents via REST API/EWS API.

## Your Environment

* Platform: PC desktop
* Host: Outlook
* Office version number: Microsoft 365 MSO (Version 2110 Build 16.0.14527.20234)
* Operating System: Windows 10 Enterprise (Version 20H2 Build 19042.1348)

**Comment by exextoc** [comment]
Thanks for reporting this issue.
We are aware of this behavior, however, we don't have any timelines to share regarding it.
Internal tracking id : Office 4714964.

**Comment by exextoc** [comment]
Would also like to add what is your scenario here? Are you trying to download the attachment contents? Or something else?

**Comment by thowoi** [comment]
Yes, it's for downloading the attachment contents. We are implementing a custom attachment encryption api.

**Comment by exextoc** [comment]
We've encountered difficulty in getting attachment ids from signed mails. We are exploring ways to enable this scenario (perhaps enable getAttachmentContentAsync) to work on these attachments, but we are still exploring this space and can't share when a solution would be implemented.

**Comment by ExternalMe** [comment]
We use this API to archive email attachments in our own cloud solution and make them accessible to our customers. Our customers trust that they can bring in any email and its attachments at any time. I am writing this comment in the hope to highlight the importance of a working interface and to encourage you, that this much needed feature will be implemented soon. It is incomprehensible to our customers that our solution does not work because Outlook offers a non-functioning interface.

I hope to hear from you guys soon.

**Comment by georgechr** [comment]
We also have this issue. It's annoying that Office JS is broken in this case

**Comment by egloff** [comment]
Same for us, hope this will be addressed soon...

**Comment by stephanmo** [comment]
We are facing the same problems. Please fix this @exextoc. MS promotes its new outlook where our old com-plugins do not work anymore and the new outlook addins are lacking such crucial functionality to read attachment contents

**Comment by Runja** [comment]
Are there any updates on this issue? Some of our customers are slowly switching to signed emails and suddenly we have to explain to them that our archiving/filing addin will no longer work properly. This is quite a hindrance for us.

**Comment by manuelsidler** [comment]
@exextoc any update on this issue?

**Comment by benjibauss** [comment]
Jup same here...any update?

**Comment by TimonGisler** [comment]
Is there any known way or workaround to access the attachments of digitally signed Outlook emails?

**Comment by stephanmo** [comment]
It has now been **almost 3** years since this issue was reported, and there is still no information on whether and when this issue will be addressed. Could you please provide an update on the timeline regarding this matter? This issue is blocking a customer's go-live, as they will be switching to the new web-based Outlook add-in and will no longer be able to use our COM plugin (which can read encrypted attachments).

**Comment by 223740** [comment]
We've encountered the same issue, Office.js is still faulty here.

**Comment by oiBio** [comment]
Another year has passed. We also have the same issue. 
Any way this will be fixed?

**Comment by Oleg-O** [comment]
Yes, unfortunately, this is still on our backlog. However, we do want to get it addressed - recently there are additional related reasons to fix this. I just don't know when we can get to it, hopefully soon.

**Comment by stephanmo** [comment]
Any update on this? It's been 4+ years. This is a critical blocker for anyone handling attachments in Outlook. When will this finally be prioritized and addressed? The community has been waiting a long time already.  Is there any serious intention to fix this at all? Or should we just accept that Office.js is broken for digitally signed/encrypted emails?
